ports/psoc6/main.c

```diff
@@ -22,38 +22,59 @@
 // port-specific includes
 #include "mplogger.h"
 
-#if MICROPY_ENABLE_COMPILER
+// port-specific includes
+#include "mplogger.h"
 
-void do_str(const char *src, mp_parse_input_kind_t input_kind) {
-    nlr_buf_t nlr;
 
-    if (nlr_push(&nlr) == 0) {
-        mp_lexer_t *lex = mp_lexer_new_from_str_len(MP_QSTR__lt_stdin_gt_, src, strlen(src), 0);
-        qstr source_name = lex->source_name;
-        mp_parse_tree_t parse_tree = mp_parse(lex, input_kind);
-        mp_obj_t module_fun = mp_compile(&parse_tree, source_name, true);
-
-        mp_call_function_0(module_fun);
-        nlr_pop();
-    } else {
-        // uncaught exception
-        mp_obj_print_exception(&mp_plat_print, (mp_obj_t)nlr.ret_val);
-    }
-}
-
-#endif
+cyhal_rtc_t psoc6_rtc;
```

Comment:
Just an abstract design level question/suggestion: 

Should we expose `cy_/hal/pdl` related macros and types to the main program? I and Nikhita also discussed about this in the end decided to encapsulate such `cy/hal_pdl` in a way defined here: https://github.com/jaenrig-ifx/micropython/blob/82a146a2f02be2ba67c5698d4a720dbc8d9d4959/ports/psoc6/drivers/psoc6_gpio.h#L18